UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -406,14 +426,35 @@ private async Task<bool> ProcessBatchFile(IEnumerable<string> uri, string downlo
                 }
 
             }
+            if (fssCacheConfiguration.Value.IsFssCacheEnabled && !entry.IgnoreCache && !entry.IsCached)
+            {
+                var productName = entry.Attributes.Where(a => a.Key == "CellName").Select(a => a.Value).FirstOrDefault();
+                var editionNumber = entry.Attributes.Where(a => a.Key == "EditionNumber").Select(a => a.Value).FirstOrDefault();
+                var updateNumber = entry.Attributes.Where(a => a.Key == "UpdateNumber").Select(a => a.Value).FirstOrDefault();
+
+                var fssSearchResponseCache = new FssSearchResponseCache
+                {
+                    BatchId = entry.BatchId,
+                    PartitionKey = productName,
+                    RowKey = $"{editionNumber}|{updateNumber}",
+                    Response = JsonConvert.SerializeObject(entry)
+                };
+                logger.LogInformation(EventIds.FileShareServiceSearchResponseStoreToCacheStart.ToEventId(), "File share service search response insert/merge request in azure table for cache started for Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with FSS BatchId:{FssBatchId}. BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", productName, editionNumber, updateNumber, entry.BatchId, queueMessage.BatchId, queueMessage.CorrelationId);
+                await fileShareServiceCache.InsertOrMergeFssCacheDetail(fssSearchResponseCache);
+                logger.LogInformation(EventIds.FileShareServiceSearchResponseStoreToCacheCompleted.ToEventId(), "File share service search response insert/merge request in azure table for cache completed for Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with FSS BatchId:{FssBatchId}. BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", productName, editionNumber, updateNumber, entry.BatchId, queueMessage.BatchId, queueMessage.CorrelationId);
```

Comment:
There is no test coverage for this new code.

Reply:
We will do this test code coverage in a separate Miscellaneous Task on Monday